program-runtime/src/accounts_data_meter.rs

```diff
@@ -26,13 +26,11 @@ impl AccountsDataMeter {
     /// Make a new AccountsDataMeter
     #[must_use]
     pub fn new(initial_accounts_data_len: u64) -> Self {
-        let accounts_data_meter = Self {
+        Self {
             maximum: MAX_ACCOUNTS_DATA_LEN,
             initial: initial_accounts_data_len,
             delta: 0,
-        };
-        debug_assert!(accounts_data_meter.initial <= accounts_data_meter.maximum);
```

Comment:
This check still seems valuable, why was it removed?

Reply:
How about we remove `maximum` from `AccountsDataMeter` as well then if it's not used?

Reply:
Excellent, looks good!